### PR TITLE
DOCS Rename SolrSearchIndex to CwpSolrIndex, update some repo URLs and code formatting for SS4

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves https://github.com/silverstripe/cwp-core/issues/4

### CODE REVIEW

File: code/pagetypes/BasePage.php [hunk 1]
```diff
@@ -282,7 +282,7 @@ class BasePage_Controller extends ContentController {
 	 */
 	public static $results_per_page = 10;
 
-	public static $search_index_class = 'SolrSearchIndex';
```
Comment (tractorcow) [inline code/pagetypes/BasePage.php]: We'll need to use the full namespaced name here. we could use `SolrSearchIndex::class`

File: docs/en/02_Features/01_Solr_search/04_Default_search_index.md [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-			// Adds 'ShowInSearch' boolean field to Solr document so we can later ensure that only documents included in 
-			// search are returned by Solr.
-			$this->addFilterField('ShowInSearch');
-
-			parent::init();
-		}
-	}
-	```
+```php
+use SilverStripe\CMS\Model\SiteTree;
```
Comment (tractorcow) [inline docs/en/02_Features/01_Solr_search/04_Default_search_index.md]: Will need extra namespace imports.

File: code/pagetypes/BasePage.php (hunk 1 above)
Comment (robbieaverill) [reply code/pagetypes/BasePage.php]: Yeah this is getting updated with the module in another PR

File: docs/en/02_Features/01_Solr_search/04_Default_search_index.md (hunk 2 above)
Comment (robbieaverill) [reply docs/en/02_Features/01_Solr_search/04_Default_search_index.md]: Added CwpSearchIndex too

File: code/pagetypes/BasePage.php (hunk 1 above)
Comment (NightJar) [reply code/pagetypes/BasePage.php]: Also `public` => `private`?
But again assuming future PR (if it's not intentional because it's not config) :)

Comment (robbieaverill) [reply code/pagetypes/BasePage.php]: yep

**Pull request merged**